Lib/importlib/resources/_common.py

```diff
@@ -113,7 +113,13 @@ def from_package(package: types.ModuleType):
     # deferred for performance (python/cpython#109829)
     from ._adapters import wrap_spec
 
-    spec = wrap_spec(package)
+    if package.__spec__ is None:
+        raise TypeError(
+            f"Cannot access resources because the code used to populate '{package.__name__}' "
+            "does not correspond directly with an importable module."
```

Comment:
Thanks, I like this version better, but I think it could be less verbose. 

```suggestion
            f"Cannot access resources for '{package.__name__}' "
            "as it does not correspond to an importable module."
```

Somebody more familiar with this code will hopefully come up with a better suggestion so maybe wait for a review from a core dev before changing.

Reply:
This sounds correct enough. I'd probably write it something like:

```python
f"Cannot access resources for '{package.__name__ or package!r}' "
"as it does appear to correspond to an importable module."
```

- `package.__name__` might be `None`, fallback to `package` in that case
- Use the repr instead of the string value for `package.__main__`, so that the name is quoted
- Don't sound so authoritative on the reason `__spec__` might be `None`, as it is possible we may be wrong

Reply:
Thank you both, I updated it!